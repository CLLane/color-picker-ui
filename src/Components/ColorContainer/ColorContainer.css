.color-container__section {
  display: flex;
  width: 100%;
  justify-content: space-around;
}