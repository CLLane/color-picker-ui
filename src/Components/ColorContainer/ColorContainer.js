import React from "react";
import ColorCard from "../ColorCard/ColorCard";
import "./ColorContainer.css";
import newColors from '../../Images/generateColors.svg';

export const ColorContainer = ({ colors, generateColors, toggleColorLock }) => {
  const colorCards = colors.map((card, index) => {
    return (
      <ColorCard 
      card={ {...card, index} } 
      toggleColorLock={toggleColorLock} 
      key={index} 
      />
    );
  });

  return (
    <div className='color-container'>
      <section className='palette-container__section'>
        {colorCards}
      </section>
      <div className='generate-color__container'>
        <img className='generate-color__image' src={newColors} alt='generate colors' onClick={generateColors}/>
        <p className='generate-colors__paragraph'>Generate Colors</p>
      </div>
    </div>
  );
};

export default ColorContainer;
