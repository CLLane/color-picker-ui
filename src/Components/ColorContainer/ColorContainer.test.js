import React from 'react';
import { shallow } from 'enzyme';
import './ColorContainer.css';
import ColorContainer from './ColorContainer';

describe('Color Container', () => {
  let wrapper;
  let mockColors;
  let mockGenerateColors;
  let mockToggleColorLock;
  beforeEach(() => {
    mockColors = [
      { color: '#FFFFFF', locked: false },
      { color: '#FFFFFF', locked: false },
      { color: '#FFFFFF', locked: false },
      { color: '#FFFFFF', locked: false },
      { color: '#FFFFFF', locked: false }
    ];
    mockGenerateColors = jest.fn();
    mockToggleColorLock = jest.fn();
    wrapper = shallow(<ColorContainer 
      colors={mockColors}
      generateColors={mockGenerateColors}
      toggleColorLock={mockToggleColorLock}
    />);
  });

  it('should match the snapshot, given a set of colors', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the generate colors function when clicked', () => {
    wrapper.find('.generate-color__container').simulate('click');
    expect(mockGenerateColors).toHaveBeenCalled();
  });
});