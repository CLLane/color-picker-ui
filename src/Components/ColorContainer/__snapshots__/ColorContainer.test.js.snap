// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Container should match the snapshot, given a set of colors 1`] = `
<div
  className="color-container"
>
  <section
    className="palette-container__section"
  >
    <ColorCard
      card={
        Object {
          "color": "#FFFFFF",
          "index": 0,
          "locked": false,
        }
      }
      key="0"
      toggleColorLock={[MockFunction]}
    />
    <ColorCard
      card={
        Object {
          "color": "#FFFFFF",
          "index": 1,
          "locked": false,
        }
      }
      key="1"
      toggleColorLock={[MockFunction]}
    />
    <ColorCard
      card={
        Object {
          "color": "#FFFFFF",
          "index": 2,
          "locked": false,
        }
      }
      key="2"
      toggleColorLock={[MockFunction]}
    />
    <ColorCard
      card={
        Object {
          "color": "#FFFFFF",
          "index": 3,
          "locked": false,
        }
      }
      key="3"
      toggleColorLock={[MockFunction]}
    />
    <ColorCard
      card={
        Object {
          "color": "#FFFFFF",
          "index": 4,
          "locked": false,
        }
      }
      key="4"
      toggleColorLock={[MockFunction]}
    />
  </section>
  <div
    className="generate-color__container"
    onClick={[MockFunction]}
  >
    <img
      alt="generate colors"
      className="generate-color__image"
      src="generateColors.svg"
    />
    <p
      className="generate-colors__paragraph"
    >
      Generate Colors
    </p>
  </div>
</div>
`;
